### ISSUE TITLE
API `ApproveMeeting` is not working

### ISSUE THREAD

**Issue by Minitour** [shalomweiss/mm-system-2017#27]
## Endpoint
```
http://zofenweb-env.hhaqjxq9wa.us-east-2.elasticbeanstalk.com:80/ApproveMeeting
```
## Request Body
```json
{
"action":true,
"meetingId":"4",
"token":"KfPTz2k2Scrj9tgIyTzmtXpUNcvo29XDndgnLUS2g5WK3PbuFY37WK3PW7wUBJmRcrpAQIYuCOIVToBIkjqB2RU6hztnRrz8VflVU0wNfrAkKuDn0wQkiEBS8hCitnNh",
"id":16
}
```

## Response
```html
<!DOCTYPE html>
<html>
    <head>
        <title>Apache Tomcat/8.0.45 - Error report</title>
        <style type="text/css">H1 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:22px;} H2 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:16px;} H3 {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;font-size:14px;} BODY {font-family:Tahoma,Arial,sans-serif;color:black;background-color:white;} B {font-family:Tahoma,Arial,sans-serif;color:white;background-color:#525D76;} P {font-family:Tahoma,Arial,sans-serif;background:white;color:black;font-size:12px;}A {color : black;}A.name {color : black;}.line {height: 1px; background-color: #525D76; border: none;}</style>
    </head>
    <body>
        <h1>HTTP Status 500 - </h1>
        <div class="line"></div>
        <p>
            <b>type</b> Exception report
        </p>
        <p>
            <b>message</b>
            <u></u>
        </p>
        <p>
            <b>description</b>
            <u>The server encountered an internal error that prevented it from fulfilling this request.</u>
        </p>
        <p>
            <b>exception</b>
        </p>
        <pre>java.lang.NullPointerException
	mm.androidservice.ApproveMeeting.doPost(ApproveMeeting.java:41)
	javax.servlet.http.HttpServlet.service(HttpServlet.java:648)
	javax.servlet.http.HttpServlet.service(HttpServlet.java:729)
	org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
</pre>
        <p>
            <b>note</b>
            <u>The full stack trace of the root cause is available in the Apache Tomcat/8.0.45 logs.</u>
        </p>
        <hr class="line">
        <h3>Apache Tomcat/8.0.45</h3>
    </body>
</html>
```

**Comment by Minitour** [comment]
I also tried 
```json
{
     "meeting_id" : "4",
  ...
}
```

**Comment by Minitour** [comment]
Found the issue:
https://github.com/shalomweiss/mm-system-2017/blob/dbaf14e961be08c255cb610e95fbcd79b3e2a64b/src/mm/androidservice/ApproveMeeting.java#L42

Why do you need `user_id` and `id`?

**Comment by Minitour** [comment]
# Notes

### Action is of type Boolean
I also noticed you fetch `action` as a String and then parse it into Integer.
https://github.com/shalomweiss/mm-system-2017/blob/dbaf14e961be08c255cb610e95fbcd79b3e2a64b/src/mm/androidservice/ApproveMeeting.java#L43

You can simply use  a boolean here. 

### Use Camel Case
We should stick to a single format, so far we have been using `Camel Case` so let's continue doing so. 
Change the constant JSON key `meeting_id` to `meetingId`.
https://github.com/shalomweiss/mm-system-2017/blob/dbaf14e961be08c255cb610e95fbcd79b3e2a64b/src/mm/androidservice/ApproveMeeting.java#L41

**Comment by firdos94** [comment]
okey, i have no idea why there is userId and id (one of them should be removed)
its not good that action is boolean, we have 3 types of meeting status:  PENDING(0),APPROVED(1),COMPLETE(2)

you can check:
Request:
{
"id":16,
"action":1,
"meetingId":4,
"token":"KfPTz2k2Scrj9tgIyTzmtXpUNcvo29XDndgnLUS2g5WK3PbuFY37WK3PW7wUBJmRcrpAQIYuCOIVToBIkjqB2RU6hztnRrz8VflVU0wNfrAkKuDn0wQkiEBS8hCitnNh"
}
to get the response:
{
    "code": 200,
    "message": "success"
}

**Comment by Minitour** [comment]
@firdos94 The `action` parameter is what indicates if we want to Approve or Decline the meeting.
If there is a third option then yes we should use an integer. But what is it?

**Comment by firdos94** [comment]
Decline type is not define as meetingtype
function in db , changemeetingstatuse accept messageStatus param, one of 3 types: pending,approve,complete

maybe we should add new type ,TOMORROW :)

**Comment by Minitour** [comment]
@NajibAbuSamra Look into this.

**Comment by Minitour** [comment]
Here is a screenshot of the work in progress `MeetingController`

<img width="240" alt="screen shot 2017-10-14 at 2 55 35" src="https://user-images.githubusercontent.com/17438617/31570318-35490340-b08b-11e7-842a-d2f3a8f8fb4e.png">

Here is what we defined in The System doc: 
<img width="275" alt="screen shot 2017-10-14 at 2 58 49" src="https://user-images.githubusercontent.com/17438617/31570368-b7b7058e-b08b-11e7-8cc7-8426f8c2a83c.png">

**Comment by NajibAbuSamra** [comment]
The WIP looks ok, I got nothing to note there.

as for the previous discussion, the way the db handles changing the meeting status is as follows: we have a function that changes a meeting's status if certain conditions are met, i.e. if a mentee is approving a meeting, then the status is changed to approved, if someone (mentor/mentee) completes a meeting then the status is changed to completed IF both sides set a completed update (that's why we have 2 columns)

In the future that same function will support change into other meeting status (declined/canceled/whatever)

**Comment by Minitour** [comment]
@firdos94, @masterFoad 

Approving a meeting works but declining a meeting results in an error.

**Comment by Minitour** [comment]
Here is the system log:
```
API-MANAGER: makeRequest: http://zofenweb-env.hhaqjxq9wa.us-east-2.elasticbeanstalk.com:80/ApproveMeeting
API-MANAGER: makeRequest: {"action":false,"token":"...","id":16,"meetingId":"3"}
API-MANAGER: onResponse: {"code":401,"message":"error"}

API-MANAGER: makeRequest: http://zofenweb-env.hhaqjxq9wa.us-east-2.elasticbeanstalk.com:80/ApproveMeeting
API-MANAGER: makeRequest: {"action":true,"token":"...","id":16,"meetingId":"3"}
API-MANAGER: onResponse: {"code":200,"message":"success"}
```

**Comment by masterFoad** [comment]
@Minitour 
on it

**Comment by masterFoad** [comment]
@dunia777

**Comment by dunia777** [comment]
DataBase said that they haven't had a declined meetingType, so we did not expect to get an action that's "false" from the request.
We will send an error message in this case untill DB create the new meetingType, Or you can make sure not to send an action that is false.

**Comment by Minitour** [comment]
@NajibAbuSamra please look into it.

**Comment by NajibAbuSamra** [comment]
I have added 2 meeting status:
DECLINED: a mentee can decline a proposed meeting
CANCELED: a mentor can cancel a meeting at any time

**Please note that in the db we do not validate that the user that sent the request is valid ** (that means a mentor that sends a declined status will return true, the validation should be done before executing the changeMeetingStatus method)

**Comment by Minitour** [comment]
@NajibAbuSamra Can mentee cancel the meeting as well? (after it was approved)

**Comment by NajibAbuSamra** [comment]
TBD, I don't see why not, though we agreed we want most of the power be with the mentor, also if memory serves there should be a message/note added in case a meeting is canceled/declined

**Comment by Minitour** [comment]
@NajibAbuSamra So this requires a new API doesn't it?

**Comment by NajibAbuSamra** [comment]
Not necessarily, we can use the function that adds mentor/mentee notes for that

**Comment by masterFoad** [comment]
@Minitour 
Can you update the ApproveMeeting request format?

**Comment by Minitour** [comment]
@masterFoad The format is still the same. If `action` is `true` then the meeting status is `APPROVED`, if it's `false` then the status will be `DECLINED`

 For the `Canceling` I will create a new issue with the API definition.

**Comment by masterFoad** [comment]
@Minitour Updated.
